protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractAuthProcessor.java

```diff
@@ -104,23 +102,23 @@ protected void doAuthWithDelegation(AuthenticationAttempt authenticationAttempt,
                     session.authenticated();
                     session.setAttribute(ImapSessionUtils.MAILBOX_SESSION_ATTRIBUTE_SESSION_KEY, mailboxSession);
                     provisionInbox(session, mailboxManager, mailboxSession);
-                    okComplete(command, tag, responder);
+                    okComplete(request, responder);
                 } catch (BadCredentialsException e) {
                     authFailure = true;
                 }
             }
             if (authFailure) {
-                manageFailureCount(session, tag, command, responder, failed);
+                manageFailureCount(session, request, responder, failed);
             }
         } catch (UserDoesNotExistException e) {
             LOGGER.info("User {} does not exist", authenticationAttempt.getAuthenticationId(), e);
-            no(command, tag, responder, HumanReadableText.USER_DOES_NOT_EXIST);
+            no(request, responder, HumanReadableText.USER_DOES_NOT_EXIST);
```

Comment:
how can you answer "no" without the tag?

Reply:
by calling the responder+response factory. That's orthogonal to this refactoring.